backend/src/router/block.ts

```diff
@@ -3,46 +3,183 @@ import { z } from "zod";
 
 import { Context } from "../context";
 import * as nearApi from "../utils/near";
-import * as blocks from "../providers/blocks";
-import * as chunks from "../providers/chunks";
-import * as receipts from "../providers/receipts";
 import { validators } from "./validators";
+import { indexerDatabase } from "../database/databases";
+import { count, div, sum } from "../database/utils";
+import { millisecondsToNanoseconds } from "../utils/bigint";
 
 export const router = trpc
   .router<Context>()
-  .query("nearcore-final-block", {
-    resolve: () => {
-      return nearApi.sendJsonRpc("block", { finality: "final" });
+  .query("final", {
```

Comment:
That looks for me like a more sensible way - as we use `nearcore-final-block` for just getting the block's height - to use it for both calls and resolve with just actual height.